test/ArcxAnalyticsSdk.test.ts

```diff
@@ -89,290 +110,460 @@ describe('(unit) ArcxAnalyticsSdk', () => {
-            ),
-        ).to.be.true
+        expect(window.ethereum?.on).calledWithMatch('accountsChanged')
+      })
+
+      it('calls _onAccountsChanged when accountsChanged is fied and trackWalletConnections is true', async () => {
+        window.ethereum = new MockEthereum() as any
+        const sdk = await ArcxAnalyticsSdk.init(TEST_API_KEY, { trackWalletConnections: true })
+
+        const onAccountsChangedStub = sinon.stub(sdk, '_onAccountsChanged' as any)
+
+        window.ethereum?.emit('accountsChanged', [TEST_ACCOUNT])
+        expect(onAccountsChangedStub).calledOnceWithExactly([TEST_ACCOUNT])
+      })
+
+      it('calls _onChainChanged when chainChanged is fired and trackChainChanges is true', async () => {
+        window.ethereum = new MockEthereum() as any
+        const sdk = await ArcxAnalyticsSdk.init(TEST_API_KEY, { trackChainChanges: true })
+
+        const onChainChangedStub = sinon.stub(sdk, '_onChainChanged' as any)
+
+        window.ethereum?.emit('chainChanged', TEST_CHAIN_ID)
+        expect(onChainChangedStub).calledOnceWithExactly(TEST_CHAIN_ID)
       })
     })
 
-    it('#event', async () => {
-      await analyticsSdk.event('TEST_EVENT', TEST_ATTRIBUTES)
-      expect(
-        postRequestStub.calledOnceWith(
-          PROD_URL_BACKEND,
+    describe('#event', () => {
+      it('calls postRequest with the correct params', async () => {
+        const testAttributes = {
+          a: 'value',
+          b: 'second value',
+        }
+
+        await analyticsSdk.event('TEST_EVENT', testAttributes)
+        expect(postRequestStub).calledOnceWith(
+          DEFAULT_SDK_CONFIG.url,
           TEST_API_KEY,
           '/submit-event',
-          getAnalyticsData('TEST_EVENT', TEST_ATTRIBUTES),
-        ),
-      ).to.be.true
-    })
-
-    it('#page', async () => {
-      const pageAttributes = { url: 'page.test' }
-      const eventStub = sinon.stub(analyticsSdk, 'event')
+          getAnalyticsData('TEST_EVENT', testAttributes),
+        )
+      })
 
-      await analyticsSdk.page(pageAttributes)
+      it('supports nested attributes', async () => {
+        const pageAttributes = { layer1: { layer2: { layer3: { layer4: 'hello!' } } } }
 
-      expect(eventStub.calledOnceWith(PAGE_EVENT, pageAttributes)).to.be.true
+        await analyticsSdk.event('TEST_EVENT', pageAttributes)
+        expect(postRequestStub).calledOnceWithExactly(
+          DEFAULT_SDK_CONFIG.url,
+          TEST_API_KEY,
+          '/submit-event',
+          getAnalyticsData('TEST_EVENT', pageAttributes),
+        )
+      })
     })
 
-    it('#connectWallet', async () => {
-      const attributes = { account: '0x12354', chain: '1' }
-      const eventStub = sinon.stub(analyticsSdk, 'event')
-
-      await analyticsSdk.connectWallet(attributes)
+    describe('#attribute', () => {
+      it('calls event() with the given attributes', async () => {
```

Comment:
You use `it('calls event() with the given attributes', async () => {` in every test. In my opinion it doesn't provide any additional information. So we can just get rid of this it and leave  `it('#[functionName', () => {`.
if you think that it provides additional sense, it is equal to situation :
```ts
describe('calls event() with the given attributes', async () => {
  it('#function1', () => {...}
  ....
   it('#functionN', () => {...}
}
```

Reply:
If you have a describe with one `it`, get rid of `describe` and only use `it`

Reply:
I considered, it, but it looks cleaner in the console when you run the tests: 
![image](https://user-images.githubusercontent.com/5834876/205325161-719d424d-f14e-4401-87ee-1c1dfd92d2c7.png)
If we don't use `describe` it makes it harder to differentiate between function tests and other tests

Reply:
ok